### PR TITLE
(Bugfix) Finalize Claim UX

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

About # (link your issue here)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/pages/gd/Claim/ClaimBalance.tsx [hunk 1]
```diff
@@ -40,13 +40,17 @@ export const ClaimBalance = () => {
                 .catch(() => undefined),
         [chainId]
     )
-    const formattedTime = useMemo(() => format(claimTime, 'hh aaa'), [claimTime])
+    const formattedTime = useMemo(() => claimTime && format(claimTime, 'hh aaa'), [claimTime])
     const { switchNetwork } = useEthers()
 
-    const network = useMemo(() => SupportedChains[chainId], [chainId])
+    const network = useMemo(
+        () => (chainId === (SupportedChains.FUSE as number) ? SupportedChains[42220] : SupportedChains[122]),
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/ClaimBalance.tsx:47]: maybe just 
```suggestion
        () => SupportedChains [chainId],
```

and do not think memo is required for so simple logic

File: src/pages/gd/Claim/ClaimBalance.tsx [hunk 2]
```diff
@@ -40,13 +40,17 @@ export const ClaimBalance = () => {
...
+        [chainId]
+    )
 
     const switchChain = useCallback(() => {
-        switchNetwork(chainId).catch(noop)
+        const toChain = chainId === (SupportedChains.FUSE as number) ? 42220 : 122
+        switchNetwork(toChain).catch(noop)
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/ClaimBalance.tsx:53]: maybe just 
```suggestion
        switchNetwork(chainId).catch(noop)
```

or even 
```suggestion
        switchNetwork(network).catch(noop)
      }, [switchNetwork, network])  
```

File: src/pages/gd/Claim/index.tsx [hunk 3]
```diff
@@ -1,17 +1,46 @@
...
+    }, [claimAmount, state, send])
+
+    const handleClaim = useCallback(async () => {
+        const claim = await send()
+        if (claim) {
+            // todo: add event analytics on transaction receipt
+            setClaimed(true)
+            return true
+        }
+        return false
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/index.tsx:41]:
```suggestion
        const claim = await send()
        
        if (!claim) {
          return false
        }
      
        // todo: add event analytics on transaction receipt
        setClaimed(true)
        return true
```

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: src/pages/gd/Claim/index.tsx [hunk 4]
```diff
@@ -33,14 +29,21 @@ const Claim = memo(() => {
...
-        return false
+
+        // todo: add event analytics on transaction receipt
+        setClaimed(true)
+        return true
     }, [send])
 
+    const handleConnect = useCallback(async () => {
+        const state = await connect()
+        return !!state.length
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/index.tsx:44]:
```suggestion

        return !!state.length
```

File: src/pages/gd/Claim/ClaimBalance.tsx [hunk 5]
```diff
@@ -40,14 +40,15 @@ export const ClaimBalance = () => {
                 .catch(() => undefined),
         [chainId]
     )
-    const formattedTime = useMemo(() => format(claimTime, 'hh aaa'), [claimTime])
+
+    const formattedTime = useMemo(() => claimTime && format(claimTime, 'hh aaa'), [claimTime])
     const { switchNetwork } = useEthers()
 
-    const network = useMemo(() => SupportedChains[chainId], [chainId])
+    const network = chainId === (SupportedChains.FUSE as number) ? SupportedChains[42220] : SupportedChains[122]
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/ClaimBalance.tsx:47]: why not just SupportedChains[chainId] ? you haven't answered

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

**Comment by johnsmith-gooddollar** [comment]
also pls resolve conflicts @L03TJ3

File: src/pages/gd/Claim/ClaimBalance.tsx (hunk 5 above)
Comment (L03TJ3) [reply src/pages/gd/Claim/ClaimBalance.tsx:47]: Because we are requesting the opposite chain, not the current chain

Comment (johnsmith-gooddollar) [reply src/pages/gd/Claim/ClaimBalance.tsx:47]: add a code comment about it

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**